nautobot/extras/api/nested_serializers.py

```diff
@@ -150,7 +158,7 @@ class Meta:
         fields = ["id", "url", "name"]
 
 
-class NestedJobLogEntrySerializer(BaseModelSerializer):
+class NestedJobLogEntrySerializer(WritableNestedSerializer):
```

Comment:
Should we allow writing/editing Nested Job Log Entries?

Reply:
`WritableNestedSerializer` is not well-named - just using it doesn't automatically make a serializer or a field writable, it just provides common infra (including the new `object_type` parameter I'm adding in this PR) for serializers that *are* declared as writable. By writable it just means that *if* you're updating a foreign object reference on the base serializer, you can identify the new foreign object either by its PK or by any unique set of fields.

The only place we actually use `NestedJobLogEntrySerializer` at present is on the `api/extras/job-results/<uuid>/logs/` endpoint, which is read-only. 

Our own docs specify that nested serializers should *always* use `WritableNestedSerializer` so this is making things align better with our own docs.

Reply:
Looking back at this, this change is not *strictly* necessary since `JobLogEntry` is not a full-featured model so it doesn't *have* to have an `object_type` field since it can't be used with Notes or ObjectChanges. But I think it's still a good change.